domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java

```diff
@@ -1764,6 +1766,47 @@ private CaseTypeEntity caseTypeEntity(String reference) {
     }
 
 
+    @Nested
+    @DisplayName("Should return SearchCriteria model whose fields match those in the SearchCriteriaEntity")
+    class MapSearchCriteriaTests {
+
+        @Test
+        void testMapSearchCriteriaEntity() {
+            SearchCriteriaEntity searchCriteriaEntity = new SearchCriteriaEntity();
+            searchCriteriaEntity.setCaseType(caseTypeEntity("CaseTypeReference"));
+            searchCriteriaEntity.setOtherCaseReference("OtherCaseReference");
+            searchCriteriaEntity.setLiveFrom(Date.from(Instant.now().minus(1, ChronoUnit.DAYS)));
+            searchCriteriaEntity.setLiveTo(Date.from(Instant.now().plus(1, ChronoUnit.DAYS)));
+
+            SearchCriteria searchCriteria = classUnderTest.map(searchCriteriaEntity);
+
+            assertEquals(searchCriteria.getCaseTypeId(), searchCriteriaEntity.getCaseType().getReference());
+            assertEquals(searchCriteria.getOtherCaseReference(), searchCriteriaEntity.getOtherCaseReference());
+            assertEquals(searchCriteria.getLiveTo(), searchCriteriaEntity.getLiveTo());
+            assertEquals(searchCriteria.getLiveFrom(), searchCriteriaEntity.getLiveFrom());
+        }
+
+        @Test
+        void testMapEmptyRoleToAccessProfileEntity() {
+            SearchCriteriaEntity searchCriteriaEntity = new SearchCriteriaEntity();
+
+            SearchCriteria searchCriteria = classUnderTest.map(searchCriteriaEntity);
+
+            assertNull(searchCriteria.getCaseTypeId());
+            assertNull(searchCriteria.getLiveFrom());
+            assertNull(searchCriteria.getLiveTo());
+            assertNull(searchCriteria.getOtherCaseReference());
+        }
+
+
+        private CaseTypeEntity caseTypeEntity(String reference) {
```

Comment:
This code is replicated in several of the nested classes within this class - could it be defined once in top level class and reused ? Appreciate its a pattern thats been copied ...

Reply:
done in [59c375d1](https://github.com/hmcts/ccd-definition-store-api/pull/1061/commits/59c375d172d81b375f2fe43039955b022afd906e)